File: iPhone/VerBank-CSTSv3-IOSClientApi/CSTS3Comm/json/CSTSPing.m
//
//  CSTSPing.m
//  JEDIv7-CSTSv3-IOSClientApi
//
//  Created by felix on 7/16/13.
//  Copyright (c) 2013 allone. All rights reserved.
//

#import "CSTSPing.h"
#import "JEDIDateTime.h"
#import "JEDISystem.h"


//
// Key define
//
static NSString * gs_CSTSPing_JsonId       = @"CSTSPing";
static NSString * gs_CSTSPing_StartTime    = @"1";
static NSString * gs_CSTSPing_EndTime      = @"2";
static NSString * gs_CSTSPing_ServerDelay  = @"3";
static NSString * gs_CSTSPing_ID           = @"4";


@implementation CSTSPing

//-----------------------------------------------------------------------------------------
- (id) init
{
    self = [super init];
    
    if(self)
    {
        [self setEntry:@"jsonId" entry:gs_CSTSPing_JsonId];
        [self setEntry:gs_CSTSPing_ID entry:[JEDISystem getUUID]];
        [self setEntry:gs_CSTSPing_StartTime entry:[JEDIDateTime currentTimeMillis]];
    }
    
    return self;
}

//-----------------------------------------------------------------------------------------
- (NSString *)  getID
{
    return [self getEntryString:gs_CSTSPing_ID];
}

//-----------------------------------------------------------------------------------------
- (void)        pingReturned
{
    JEDI_SYS_Try
    {
        long long endTime = [JEDIDateTime currentMillis] - [self getServerDelay];
        [self setEntry:gs_CSTSPing_EndTime entry:[NSNumber numberWithLongLong:endTime]];
    }
    JEDI_SYS_EndTry
}

//-----------------------------------------------------------------------------------------
- (long)        getPing
{
    return (long)([self getEndTime] - [self getStartTime]);
}

//-----------------------------------------------------------------------------------------
- (long long)    getStartTime
{
    return [self getEntryLongLong:gs_CSTSPing_StartTime];
}

//-----------------------------------------------------------------------------------------
- (void)        setStartTime:(long long) time
{
    [self setEntry:gs_CSTSPing_StartTime entry:[NSNumber numberWithLongLong:time]];
}

//-----------------------------------------------------------------------------------------
- (long long)    getEndTime
{
    return [self getEntryLongLong:gs_CSTSPing_EndTime];
}

//-----------------------------------------------------------------------------------------
- (long)        getServerDelay
{
    return [self getEntryLong:gs_CSTSPing_ServerDelay];
}

//-----------------------------------------------------------------------------------------
- (void)        setServerDelay:(long) delay
{
    [self setEntry:gs_CSTSPing_StartTime entry:[NSNumber numberWithLong:delay]];
}

@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/CSTS3/comm/jsondata/CSTSPing.java
package jedi.ex01.CSTS3.comm.jsondata;

import java.util.UUID;

import allone.json.AbstractJsonData;

public class CSTSPing extends AbstractJsonData{

	public static final String jsonId = "CSTSPing";

	public static final String startTime = "1";
	public static final String endTime = "2";
	public static final String serverDelay = "3";
	public static final String ID = "4";

	public CSTSPing() {
		super();
		setEntry("jsonId", jsonId);
		setEntry(CSTSPing.ID, UUID.randomUUID().toString());
		setEntry(CSTSPing.startTime, System.currentTimeMillis());
	}

	public String getID() {
		try {
			String data=getEntryString(CSTSPing.ID);
			return data;
		} catch (RuntimeException e) {
			return null;
		}
	}

	public void pingReturned() {
		setEntry(CSTSPing.endTime, System.currentTimeMillis()+getServerDelay());
	}

	public long getPing() {
		return getEndTime() - getStartTime();
	}

	public long getStartTime() {
		try {
			Long data=getEntryLong(CSTSPing.startTime);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setStartTime(long startTime) {
		setEntry(CSTSPing.startTime, startTime);
	}

	public long getEndTime() {
		try {
			Long data=getEntryLong(CSTSPing.endTime);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public long getServerDelay() {
		try {
			Long data=getEntryLong(CSTSPing.serverDelay);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setServerDelay(long serverDelay) {
		setEntry(CSTSPing.serverDelay, serverDelay);
	}
	
}
